Makefile

```diff
@@ -1,11 +1,52 @@
-BUILD_DIR=./build
+SHELL := /bin/bash
 
-clean:
-	rm -rf ${BUILD_DIR}
+# Env var inputs for container image builds
+# REGISTRY: The registry to which the build image should be pushed to.  Defaults to "docker.io" (Docker Hub)
+# IMAGE: The name of the image to build and publish in the afore mentioned registry.  Defaults to "gotify/cli"
+# PLATFORM: The platform for which the image should be built, defaults to amd64, arm64, i386, arm/v7 and riscv64
+REGISTRY ?= docker.io
+IMAGE ?= gotify/cli
+PLATFORM ?= linux/amd64,linux/arm64,linux/386,linux/arm/v7,linux/riscv64
+
+# Env var inputs for all builds
+# VERSION: The version for which the container image or the binary is being built.  This variable has not default.
+#          When it is not provided, no version will be specified in the built package.
+# COMMIT: The commit of this project for which the cli is being built, for reference in the tool's "version" command.
+#         Default to git's HEAD
+# LD_FLAGS: Build flags, for the tool's "version" command.  Defaults to current date, commit and version (if any).
+COMMIT ?= $(shell git rev-parse --verify HEAD)
+LD_FLAGS ?= $(if $(VERSION),-X main.Version=${VERSION}) \
+	-X main.BuildDate=$(shell date "+%F-%T") \
+	-X main.Commit=${COMMIT}
+
+# Resolve go version
+ifdef GOTOOLCHAIN
+	GO_VERSION=$(GOTOOLCHAIN)
+else
+	GO_VERSION=$(shell go mod edit -json | jq -r .Toolchain | sed -e 's/go//')
+endif
+DOCKER_GO_BUILD=go build -mod=readonly -a -installsuffix cgo -ldflags "$$LD_FLAGS"
+
+build-docker-multiarch:
+	docker buildx build \
+		--sbom=true \
+		--provenance=true \
+		$(if $(DOCKER_BUILD_PUSH),--push) \
+		-t ${REGISTRY}/${IMAGE}:${COMMIT} \
```

Comment:
I'll change this to `:master`, this repo will already have releases for nearly all commits, so we don't have to have a separate tag for this.